src/Api/JsonClient.php

```diff
@@ -108,34 +86,14 @@ public function __call($name, array $params)
      */
     protected function postRequest(array $data)
     {
-        $port = $this->useSsl ? 443 : 80;
-        $host = ($this->useSsl ? 'ssl://' : '') . $this->host;
-
-        $socket = @fsockopen($host, $port);
-        if ($socket === false) {
-            $error = error_get_last();
-            throw new \RuntimeException("Failed to connect to {$this->host} on port $port, {$error['message']}");
-        }
+        $socket = $this->getConnection();
```

Comment:
Want to keep the main API client not relying on extensions. It's why `composer.json` only suggests cURL, rather than require it.